ReimaginedLauncher/HttpClients/NexusModsHttpClient.cs

```diff
@@ -34,6 +34,22 @@ public NexusModsHttpClient(HttpClient httpClient)
         var stream = await response.Content.ReadAsStreamAsync();
         return await JsonSerializer.DeserializeAsync<NexusModsFileListResponse>(stream, SerializerOptions.CamelCase);
     }
+    
+    public async Task<NexusModsFileListResponse?> GenerateDownloadLink(string gameName, int modid, int fileId)
+    {
+        await FindAndSetApiKey();
+        var url = $"{BaseUrl}/games/{gameName}/mods/{modid}/files/{fileId}/download_link.json";
+        var response = await _httpClient.GetAsync(url);
+
+        if (!response.IsSuccessStatusCode)
+        {
+            await Console.Error.WriteLineAsync($"Failed to generate download link: {response.StatusCode}");
+            return null;
+        }
+
+        var stream = await response.Content.ReadAsStreamAsync();
+        return await JsonSerializer.DeserializeAsync<NexusModsFileListResponse>(stream, SerializerOptions.CamelCase);
```

Comment:
This is not the correct response type for this endpoint
![image](https://github.com/user-attachments/assets/390687fd-5173-4e0b-8426-9909714284ee)
Needs a model created